### DISCUSSION TITLE
Does Training an existing model cause the model to 'forget'?

### DISCUSSION THREAD

**Discussion by farrandi** [explosion/spaCy#8331] [Help: Coding & Implementations] [answered]
I asked a previous discussion thread: https://github.com/explosion/spaCy/discussions/8323
To recap: I wanted to train a NER from an existing scispacy model and I ended up just changing the config file:
```
[components.ner]
source = "en_core_sci_sm"

[components.tok2vec]
source = "en_core_sci_sm"
```
Those are the only things I changed in the config.cfg file instead of the usual `factory =` 

However, after training the code I checked it with this code :
```python:
nlp = spacy.blank('en')
print("blank:", len(nlp.vocab))
nlp = spacy.load(r"./output/Trial/model-best")
print("trained:", len(nlp.vocab))

nlp = spacy.load("en_core_sci_sm")
print("untrained sci:", len(nlp.vocab))
nlp = spacy.load(r"./output/scispacy1/model-best")
print("trained sci:", len(nlp.vocab))
```
and I got:
blank: 761
trained: 761
untrained sci: 799
trained sci: 761

Is this supposed to be right? Why is the vocab length of the trained model the same from a blank one?
Is the blank one supposed to have any vocabs at all? should it not be 0?
and how did my retrained model from scispacy have less vocabs than the original scispacy?

Is there a way to check that I did indeed retrain from a scispacy model?
Thanks!

**Comment by polm** [comment, marked as answer]
> Is the blank one supposed to have any vocabs at all? should it not be 0?

The vocab of blank models is filled with words spaCy uses internally, like part of speech names and so on. So it should not be zero.

> Is this supposed to be right? Why is the vocab length of the trained model the same from a blank one?

The vocab length is not what you think it is, and is not generally a useful number. There's some discussion of this [here](https://stackoverflow.com/questions/59515740/how-to-find-the-vocabulary-size-of-a-spacy-model) (see the notes on 2.3+), but the length of the vocab is mostly an implementation detail.

`len(vocab.strings)` is probably closer to what you want. 

> Is there a way to check that I did indeed retrain from a scispacy model?

Does your model give you good results? That's the most important thing and easiest to check directly.

Models don't save their history, so all you end up with is weights. If you don't trust the output of the training command and you want to see that it's using the scispacy model, try training for a very small number of iterations and comparing the model to the original scispacy model - you should see it hasn't changed much.

**Reply by farrandi** [reply]
Thank you so much!
